spec/models/repository/git_spec.rb

```diff
@@ -0,0 +1,362 @@
+        expect(rev_a_commit.scmid).to eq('4f26664364207fa8b1af9f8722647ab2d4ac5d43')
+        # Mon Jul 5 22:34:26 2010 +0200
+        committed_on = Time.gm(2010, 7, 5, 20, 34, 26)
+        expect(rev_a_commit.committed_on).to eq(committed_on)
+        latest_rev = instance.latest_changeset
+        expect(instance.latest_changeset.committed_on).to eq(committed_on)
+
+        instance.fetch_changesets
+        expect(instance.changesets.count).to eq(21)
+      end
+
+      describe '.latest_changesets' do
+        it 'should fetch changesets with limits' do
+          changesets = instance.latest_changesets('', nil, 2)
+          expect(changesets.size).to eq(2)
+        end
+
+        it 'should fetch changesets with paths' do
+          changesets = instance.latest_changesets('images', nil)
+          expect(changesets.map(&:revision))
+            .to eq([
+                  'deff712f05a90d96edbd70facc47d944be5897e3',
+                  '899a15dba03a3b350b89c3f537e4bbe02a03cdc9',
+                  '7234cb2750b63f47bff735edc50a1c0a433c2518'
+                ])
+
+          changesets = instance.latest_changesets('README', nil)
+          expect(changesets.map(&:revision))
+            .to eq([
+                  '32ae898b720c2f7eec2723d5bdd558b4cb2d3ddf',
+                  '4a07fe31bffcf2888791f3e6cbc9c4545cefe3e8',
+                  '713f4944648826f558cf548222f813dabe7cbb04',
+                  '61b685fbe55ab05b5ac68402d5720c1a6ac973d1',
+                  '899a15dba03a3b350b89c3f537e4bbe02a03cdc9',
+                  '7234cb2750b63f47bff735edc50a1c0a433c2518'
+                ])
+        end
+        
+        it 'should fetch changesets with path, revision and limit' do
+          changesets = instance.latest_changesets('images', '899a15dba')
+          expect(changesets.map(&:revision))
+            .to eq([
+                  '899a15dba03a3b350b89c3f537e4bbe02a03cdc9',
+                  '7234cb2750b63f47bff735edc50a1c0a433c2518'
+                ])
+
+          changesets = instance.latest_changesets('images', '899a15dba', 1)
+          expect(changesets.map(&:revision))
+            .to eq(['899a15dba03a3b350b89c3f537e4bbe02a03cdc9'])
+
+          changesets = instance.latest_changesets('README', '899a15dba')
+          expect(changesets.map(&:revision))
+            .to eq([
+                  '899a15dba03a3b350b89c3f537e4bbe02a03cdc9',
+                  '7234cb2750b63f47bff735edc50a1c0a433c2518'
+                  ])
+
+          changesets = instance.latest_changesets('README', '899a15dba', 1)
+          expect(changesets.map(&:revision))
+            .to eq(['899a15dba03a3b350b89c3f537e4bbe02a03cdc9'])
+        end
+
+        it 'should fetch changesets with tag' do
+          changesets = instance.latest_changesets('images', 'tag01.annotated')
+          expect(changesets.map(&:revision))
+            .to eq([
+                  '899a15dba03a3b350b89c3f537e4bbe02a03cdc9',
```

Comment:
Use 2 spaces for indentation in an array, relative to the start of the line where the left bracket is.